vmdb/app/models/mixins/naming_index_mixin.rb

```diff
@@ -0,0 +1,20 @@
+module NamingIndexMixin
+  extend ActiveSupport::Concern
+
+  NAMING = 'naming'
+
+  included do
+    has_many :naming_indexes, :as => :resource, :dependent => :destroy, :class_name => "CustomAttribute", :conditions => {:source => NAMING}
+  end
+
+  def next_naming_index(name)
+    with_lock do
+      record = naming_indexes.find_or_create_by_name(name)
+      record.source = NAMING
+      record.value = record.value ? record.value.to_i + 1 : 1
```

Comment:
If the record value is nil at the start, then you can just to `record.value = record.value.to_i + 1`